lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -136,177 +145,262 @@ public void OnNext(IDriverStarted value)
-                        "User has not given any way to handle IMRU result, defaulting to ignoring it");
-                }
-
+                Logger.Log(Level.Verbose, "Submitting master task");
+                var groupCommConfig = GetGroupCommConfiguration();
                 _commGroup.AddTask(IMRUConstants.UpdateTaskName);
-                _groupCommTaskStarter.QueueTask(partialTaskConf, activeContext);
+                var clientTaskConfig = Configurations.Merge(groupCommConfig, GetUpdateTaskConfiguration());
+                _groupCommTaskStarter.QueueTask(clientTaskConfig, activeContext);
+                RequestMapEvaluators(_totalMappers);
             }
             else
             {
-                string taskId;
-
-                if (!_taskIdStack.TryPop(out taskId))
-                {
-                    Logger.Log(Level.Warning, "No task Ids exist for the active context {0}. Disposing the context.",
-                        activeContext.Id);
-                    activeContext.Dispose();
-                    return;
-                }
-
-                IConfiguration mapSpecificConfig;
-
-                if (!_perMapperConfiguration.TryPop(out mapSpecificConfig))
-                {
-                    Logger.Log(Level.Warning,
-                        "No per map configuration exist for the active context {0}. Disposing the context.",
-                        activeContext.Id);
-                    activeContext.Dispose();
-                    return;
-                }
-
-                var partialTaskConf =
-                    TangFactory.GetTang()
-                        .NewConfigurationBuilder(new[]
-                        {
-                            TaskConfiguration.ConfigurationModule
-                                .Set(TaskConfiguration.Identifier, taskId)
-                                .Set(TaskConfiguration.Task, GenericType<MapTaskHost<TMapInput, TMapOutput>>.Class)
-                                .Build(),
-                            _configurationManager.MapFunctionConfiguration,
-                            mapSpecificConfig
-                        })
-                        .BindNamedParameter(typeof(InvokeGC), _invokeGC.ToString())
-                        .Build();
-
+                Logger.Log(Level.Verbose, "Submitting map task");
+                _serviceAndContextConfigurationProvider.RecordActiveContextPerEvaluatorId(activeContext.EvaluatorId);
+                var groupCommConfig = GetGroupCommConfiguration();
+                string taskId = string.Format("{0}-{1}-Version0",
+                    IMRUConstants.MapTaskPrefix,
+                    _serviceAndContextConfigurationProvider.GetPartitionIdByEvaluatorId(activeContext.EvaluatorId));
                 _commGroup.AddTask(taskId);
-                _groupCommTaskStarter.QueueTask(partialTaskConf, activeContext);
+                var clientTaskConfig = Configurations.Merge(groupCommConfig,
```

Comment:
Same as above: Can't `GetMapTaskConfiguration()` include the configuration provided by `GetGroupCommConfiguration()`?